blockchain/stake/staketx.go

```diff
@@ -697,9 +697,9 @@ func VerifyStakingPkhsAndAmounts(
 		}
 	}
 
+	var rule bool
```

Comment:
This should be inside the for loop scope imo since it's not needed outside of that scope.